pkg/cmd/cli/cmd/set/volume.go

```diff
@@ -285,10 +289,20 @@ func (a *AddVolumeOptions) Validate(isAddOp bool) error {
 					return errors.New("must provide --path for --type=hostPath")
 				}
 			case "secret":
+				if len(a.DefaultMode) > 0 {
+					if ok, _ := regexp.MatchString(`\b0?[0-7]{3}\b`, a.DefaultMode); !ok {
```

Comment:
I think we'd have to check it before we convert it into an int32 anyways, since the user could pass any string. I'll add a check for non-secret/configMap.